### PR TITLE
Send update EDS when CDS has changed

### PR DESCRIPTION
**Pull request by lukidzi**
https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#resource-warming

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt [hunk 1]
```diff
@@ -28,18 +28,23 @@ internal class SnapshotsVersions {
...
+    /**
+     * When cluster change we should also send EDS.
+     */
     private fun selectEndpoints(
         previous: VersionsWithData,
-        endpoints: List<ClusterLoadAssignment>
-    ) = if (previous.endpoints == endpoints) previous.version.endpoints else EndpointsVersion(newVersion())
+        endpoints: List<ClusterLoadAssignment>,
+        clusterChanged: Boolean
+    ) = if (previous.endpoints == endpoints && !clusterChanged)
```
Comment (MarcinFalkowski) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt]: Maybe switch to: ` !clusterChanged && previous.endpoints == endpoints` to not compare endpoints when clusters changed

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt [hunk 2]
```diff
@@ -3,7 +3,7 @@ package pl.allegro.tech.servicemesh.envoycontrol.snapshot
 import io.envoyproxy.envoy.api.v2.Cluster
 import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment
 import pl.allegro.tech.servicemesh.envoycontrol.groups.Group
-import java.util.UUID
+import java.util.*
```
Comment (slonka) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt]: I think that some people on the team were against star imports and there is a rule in ktlint about that: https://circleci.com/gh/allegro/envoy-control/425?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

You can configure your intellij to not use star imports:
image

Just check "when at least" and set it to 9999

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt (hunk 1 above)
Comment (lukidzi) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt]: Done

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt (hunk 2 above)
Comment (lukidzi) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotsVersions.kt]: Done

**Pull request merged**